libraries/AP_Proximity/AP_Proximity.cpp

```diff
@@ -367,6 +368,14 @@ void AP_Proximity::detect_instance(uint8_t instance)
         drivers[instance] = new AP_Proximity_AirSimSITL(*this, state[instance]);
         return;
 
+#endif
+
+#if HAL_MAX_CAN_PROTOCOL_DRIVERS
```

Comment:
```suggestion
#if AP_PROXIMITY_ARS408_ENABLED
```

... which can default to something based on `HAL_MAX_CAN_PROOTCOL_DRIVERS`